code/renderers/vue3/src/decorateStory.ts

```diff
@@ -25,18 +25,24 @@ function prepare(
     return null;
   }
 
+  // TODO: this doesn't work at all with decorators. they don't have args on setup()
+  // @ts-expect-error types are strict here, this will still work in the majority of cases where users don't have arguments in their setup() functions
+  const args = story.setup?.().args;
```

Comment:
This only works because we're instructing users to bind args in their stories like this:
```js
// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args) => ({
  // Components used in your story `template` are defined in the `components` object
  components: { MyButton },
  // The story's `args` need to be mapped into the template through the `setup()` method
  setup() {
    return { args };
  },
  // And then the `args` are bound to your component with `v-bind="args"`
  template: '<my-button v-bind="args" />',
});
```
So we can run the `setup()` function to read the args. However this won't work if there are decorators, because they don't follow the same pattern, they might just call `storyFn()` with a `template`.

A few unknowns from me: 
1. Usually `setup()` is invoked by the Vue renderer with VueContext as an argument, so if any user relies on that it would break here since we don't supply that argument. Does users do that?
2. This assumes `setup()` has no side-effects and we can run it whenever we like. can we?


Ideally there would be a completely other way to access the __props__ as this here feels very hacky, but I haven't found it yet.

Reply:
When you refer to side effects and re-running setup, do you mean in the story itself or a different setup function in one of the dependent functions?

I think given the current documentation, setup is the obvious place to add any necessary state to a component as it is already there, and it for now it will have to be done to demonstrate how any component works because reactivity is fundamentally broken with controls.

(This all seems especially strange given that we didn't have this problem with Vue 2 and the whole point of Vue 3 is being able to track reactive variables outside of components... I know it's related to how storybook functions deep down, so it also feels like the best solve might be outside the vue addon itself)